### PR TITLE
(EVENT) The Weave Resurgence

### PR DESCRIPTION
**Pull request by realforest2001**
# EVENT TM

### CODE REVIEW

File: code/__DEFINES/chemistry.dm [hunk 1]
```diff
@@ -93,6 +93,8 @@
 #define PLASMA_EGG "eggplasma"
 #define PLASMA_NEUROTOXIN "neurotoxinplasma"
 #define PLASMA_ROYAL "royalplasma"
+#define PLASMA_WEAVE			"weaveplasma"
+#define PLASMA_WEAVE_EXALTED	"weaveplasmaex"
```
Comment (harryob) [inline code/__DEFINES/chemistry.dm:97]:
```suggestion
#define PLASMA_WEAVE	"weaveplasma"
#define PLASMA_WEAVE_EXALTED "weaveplasmaex"
```

File: code/__DEFINES/chemistry.dm [hunk 2]
```diff
@@ -205,6 +207,7 @@
 #define PROPERTY_RADIUS "radius" // Fire related admin property
 #define PROPERTY_INTENSITY "intensity" // Fire related admin property
 #define PROPERTY_DURATION "duration" // Fire related admin property
+#define PROPERTY_WEAVING			"weaving" //Enslaves to The Weave
```
Comment (harryob) [inline code/__DEFINES/chemistry.dm:210]:
```suggestion
#define PROPERTY_WEAVING	"weaving" //Enslaves to The Weave
```

File: code/__DEFINES/mobs.dm [hunk 3]
```diff
@@ -370,6 +376,9 @@
 // Hellhound strain flags
 #define HELLHOUND_NORMAL "Normal"
 
+// Weave strain flags
+#define WEAVE_NORMAL	"Normal"
```
Comment (harryob) [inline code/__DEFINES/mobs.dm:380]:
```suggestion
#define WEAVE_NORMAL "Normal"
```

File: code/datums/diseases/weave_mind.dm [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+				to_chat(affected_mob, pick(SPAN_XENODANGER("The Weave calls..."), SPAN_XENODANGER("Follow The Weave...")))
+				affected_mob.adjustBrainLoss(5)
+		if(5)
+			affected_mob.adjustBrainLoss(10)
+			affected_mob.updatehealth()
+			if(prob(40))
+				to_chat(affected_mob, SPAN_XENOHIGHDANGER("Images flash through your mind as The Weave makes itself known..."))
+				var/mob/living/carbon/human/H = affected_mob
+				H.WeaveClaim(CAUSE_ESSENCE)
+				src.cure()
```
Comment (harryob) [inline code/datums/diseases/weave_mind.dm:37]:
```suggestion
				cure()
```

File: code/game/objects/items/weapons/blades.dm [hunk 5]
```diff
@@ -229,3 +233,28 @@
...
+	unacidable = TRUE
+
+/obj/item/weapon/weaveblade/IsShield()
+	return TRUE
+
+/obj/item/weapon/weaveblade/attack_hand(mob/living/carbon/user) //Prevents marines from instantly picking it up via pickup macros.
+	if(!((user.hivenumber == XENO_HIVE_WEAVE) || (user.faction == "The Weave Nexus")))
+		user.visible_message(SPAN_BOLDWARNING("[user] is repelled from \the [src] by the force of The Weave!"), SPAN_BOLDWARNING("The Weave repels your hand away from \the [src]! You cannot touch it!"))
+		return FALSE
+	else ..()
```
Comment (harryob) [inline code/game/objects/items/weapons/blades.dm:247]:
```suggestion
	else 
		return ..()
```

File: code/game/objects/items/weapons/blades.dm [hunk 6]
```diff
@@ -229,3 +233,28 @@
...
+
+/obj/item/weapon/weaveblade/attack(mob/living/target as mob, mob/living/carbon/human/user as mob)
```
Comment (harryob) [inline code/game/objects/items/weapons/blades.dm:249]:
```suggestion
/obj/item/weapon/weaveblade/attack(mob/living/target, mob/living/carbon/human/user)
```

File: code/game/objects/items/weapons/blades.dm [hunk 7]
```diff
@@ -229,3 +233,28 @@
...
+
+	indestructible = TRUE
+	unacidable = TRUE
+
+/obj/item/weapon/weaveblade/IsShield()
+	return TRUE
+
+/obj/item/weapon/weaveblade/attack_hand(mob/living/carbon/user) //Prevents marines from instantly picking it up via pickup macros.
+	if(!((user.hivenumber == XENO_HIVE_WEAVE) || (user.faction == "The Weave Nexus")))
+		user.visible_message(SPAN_BOLDWARNING("[user] is repelled from \the [src] by the force of The Weave!"), SPAN_BOLDWARNING("The Weave repels your hand away from \the [src]! You cannot touch it!"))
```
Comment (harryob) [inline code/game/objects/items/weapons/blades.dm:245]:
```suggestion
		user.visible_message(SPAN_BOLDWARNING("[user] is repelled from [src] by the force of The Weave!"), SPAN_BOLDWARNING("The Weave repels your hand away from [src]! You cannot touch it!"))
```

File: code/game/objects/items/weapons/blades.dm [hunk 8]
```diff
@@ -229,3 +233,28 @@
...
+/obj/item/weapon/weaveblade/attack(mob/living/target as mob, mob/living/carbon/human/user as mob)
+	if(target.hivenumber == XENO_HIVE_WEAVE || target.faction == "The Weave Nexus")
+		to_chat(user, SPAN_XENOHIGHDANGER("You cannot harm a being of The Weave!"))
+		return FALSE
+	..()
+
+/obj/item/weapon/weaveblade/pull_response(mob/puller)
+	var/mob/living/carbon/user = puller
+	if(!((user.hivenumber == XENO_HIVE_WEAVE) || (user.faction == "The Weave Nexus")))
+		user.visible_message(SPAN_BOLDWARNING("[user] is repelled from \the [src] by the force of The Weave!"), SPAN_BOLDWARNING("The Weave repels your hand away from \the [src]! You cannot touch it!"))
```
Comment (harryob) [inline code/game/objects/items/weapons/blades.dm:258]:
```suggestion
		user.visible_message(SPAN_BOLDWARNING("[user] is repelled from [src] by the force of The Weave!"), SPAN_BOLDWARNING("The Weave repels your hand away from [src]! You cannot touch it!"))
```

File: code/modules/gear_presets/other.dm [hunk 9]
```diff
@@ -971,3 +971,71 @@
 	H.equip_to_slot_or_del(new /obj/item/storage/pouch/tools/tank(H), WEAR_R_STORE)
 
 //*****************************************************************************************************/
+
+
+
+
+
+/datum/equipment_preset/other/weave_cultist
```
Comment (harryob) [inline code/modules/gear_presets/other.dm:979]:
```suggestion
/datum/equipment_preset/other/weave_cultist
```

File: code/modules/mob/living/carbon/human/human_abilities.dm [hunk 10]
```diff
@@ -607,3 +607,136 @@ CULT
 
 	var/mob/living/carbon/human/human_user = owner
 	SEND_SIGNAL(human_user, COMSIG_MOB_MG_EXIT)
+
+
+
+//Weave
+/datum/action/human_action/activable/cult/weave/speak_hivemind
```
Comment (harryob) [inline code/modules/mob/living/carbon/human/human_abilities.dm:614]:
```suggestion
//Weave
/datum/action/human_action/activable/cult/weave/speak_hivemind
```

File: code/modules/mob/living/carbon/human/human_abilities.dm [hunk 11]
```diff
@@ -607,3 +607,136 @@ CULT
...
+		return FALSE
+
+	if(nexus.use_energy(300))
+		enter_cooldown(cooldown)
+		nexus.bioscan_time = world.time + 15 MINUTES
+		to_chat(owner, SPAN_XENONOTICE("The Weave reveals there are [marines] humans and [xenos] other biosigns alive on the colony."))
+		return TRUE
+	else
+		to_chat(owner, SPAN_XENOWARNING("The Weave is not strong enough here to do that!"))
+		return FALSE
```
Comment (harryob) [inline code/modules/mob/living/carbon/human/human_abilities.dm:677]:
```suggestion
		return TRUE

	to_chat(owner, SPAN_XENOWARNING("The Weave is not strong enough here to do that!"))
	return FALSE
```

File: code/modules/mob/living/carbon/human/human_abilities.dm [hunk 12]
```diff
@@ -607,3 +607,136 @@ CULT
...
+	if (do_after(cultist, 10 SECONDS, INTERRUPT_ALL | BEHAVIOR_IMMOBILE, BUSY_ICON_FRIENDLY))
+		if (cultist.blood_volume < BLOOD_VOLUME_SAFE)
+			to_chat(cultist, SPAN_XENOWARNING("You do not have enough blood to do this safely! The Weave will not risk harming you."))
+			return FALSE
+		else
+			cultist.blood_volume -= 100
+			nexus.weave_energy += 100
+			cultist.visible_message(SPAN_XENONOTICE("[cultist] exudes energy back into The Weave!"), SPAN_XENONOTICE("You release some of your energy into The Weave!"))
+			enter_cooldown(cooldown)
+			return TRUE
```
Comment (harryob) [inline code/modules/mob/living/carbon/human/human_abilities.dm:708]:
```suggestion
			return FALSE

		cultist.blood_volume -= 100
		nexus.weave_energy += 100
		cultist.visible_message(SPAN_XENONOTICE("[cultist] exudes energy back into The Weave!"), SPAN_XENONOTICE("You release some of your energy into The Weave!"))
		enter_cooldown(cooldown)
		return TRUE
```

File: code/modules/mob/living/carbon/human/human_abilities.dm [hunk 13]
```diff
@@ -607,3 +607,136 @@ CULT
...
+	var/datum/hive_status/mutated/weave/nexus = GLOB.hive_datum[XENO_HIVE_WEAVE]
+	if(!istype(nexus))
+		to_chat(owner, SPAN_WARNING("Error: Weave Hive not found. Ping forest2001."))
+		return FALSE
+	var/mob/living/carbon/human/cultist = owner
+	if (cultist.blood_volume < BLOOD_VOLUME_SAFE)
+		to_chat(cultist, SPAN_XENOWARNING("You do not have enough blood to do this safely! The Weave will not risk harming you."))
+		return FALSE
+
+	if (do_after(cultist, 10 SECONDS, INTERRUPT_ALL | BEHAVIOR_IMMOBILE, BUSY_ICON_FRIENDLY))
```
Comment (harryob) [inline code/modules/mob/living/carbon/human/human_abilities.dm:699]: early return

File: code/modules/mob/living/carbon/human/human_abilities.dm [hunk 14]
```diff
@@ -607,3 +607,136 @@ CULT
...
+		return FALSE
+	var/mob/living/carbon/human/cultist = owner
+	if (!nexus.can_use_energy(energy_cost))
+		to_chat(cultist, SPAN_XENOWARNING("The Weave is not strong enough here to do that!"))
+		return FALSE
+	if(cultist.bruteloss < 10 && cultist.fireloss < 10)
+		to_chat(cultist, SPAN_XENOWARNING("You are not sufficiently injured for The Weave to aid you."))
+		return FALSE
+
+	if (do_after(cultist, 10 SECONDS, INTERRUPT_ALL | BEHAVIOR_IMMOBILE, BUSY_ICON_MEDICAL))
```
Comment (harryob) [inline code/modules/mob/living/carbon/human/human_abilities.dm:733]: early return

File: code/modules/mob/living/carbon/human/human_abilities.dm [hunk 15]
```diff
@@ -607,3 +607,136 @@ CULT
...
+	var/energy_cost = 150
+
+/datum/action/human_action/activable/cult/weave/regenerate_wounds/action_activate()
+	if(!can_use_action())
+		return FALSE
+	if(!action_cooldown_check())
+		return FALSE
+	var/datum/hive_status/mutated/weave/nexus = GLOB.hive_datum[XENO_HIVE_WEAVE]
+	if(!istype(nexus))
+		to_chat(owner, SPAN_WARNING("Error: Weave Hive not found. Ping forest2001."))
```
Comment (harryob) [inline code/modules/mob/living/carbon/human/human_abilities.dm:723]: glare

File: code/modules/mob/living/carbon/human/human_abilities.dm [hunk 16]
```diff
@@ -607,3 +607,136 @@ CULT
...
+	cooldown = 3 MINUTES
+
+/datum/action/human_action/activable/cult/weave/exude_energy/action_activate()
+	if(!can_use_action())
+		return FALSE
+	if(!action_cooldown_check())
+		return FALSE
+	var/datum/hive_status/mutated/weave/nexus = GLOB.hive_datum[XENO_HIVE_WEAVE]
+	if(!istype(nexus))
+		to_chat(owner, SPAN_WARNING("Error: Weave Hive not found. Ping forest2001."))
```
Comment (harryob) [inline code/modules/mob/living/carbon/human/human_abilities.dm:692]: glare

File: code/modules/mob/living/carbon/human/human_abilities.dm [hunk 17]
```diff
@@ -607,3 +607,136 @@ CULT
...
+	if (do_after(cultist, 10 SECONDS, INTERRUPT_ALL | BEHAVIOR_IMMOBILE, BUSY_ICON_MEDICAL))
+		if(!nexus.use_energy(energy_cost))
+			to_chat(cultist, SPAN_XENOWARNING("The Weave is not strong enough here to do that!"))
+			return FALSE
+		else
+			cultist.visible_message(SPAN_XENONOTICE("[cultist] is wrapped in energies from The Weave as their wounds close!"), SPAN_XENONOTICE("Your wounds close as The Weave wraps around you!"))
+			cultist.apply_damage(-20, BRUTE)
+			cultist.apply_damage(-20, BURN)
+			enter_cooldown(cooldown)
+			return TRUE
```
Comment (harryob) [inline code/modules/mob/living/carbon/human/human_abilities.dm:742]:
```suggestion
		cultist.visible_message(SPAN_XENONOTICE("[cultist] is wrapped in energies from The Weave as their wounds close!"), SPAN_XENONOTICE("Your wounds close as The Weave wraps around you!"))
		cultist.apply_damage(-20, BRUTE)
		cultist.apply_damage(-20, BURN)
		enter_cooldown(cooldown)
		return TRUE
```

File: code/modules/mob/living/carbon/xenomorph/abilities/TheWeave/prime_powers.dm [hunk 18]
```diff
@@ -0,0 +1,99 @@
...
+	var/mob/living/carbon/xenomorph/prime_weaver/self = owner
+
+	if (!action_cooldown_check())
+		return FALSE
+
+	if (!self.check_state())
+		return FALSE
+
+	var/mob/living/carbon/human/human = target
+	if	(!ishuman(human) || !human.allow_gun_usage)
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/abilities/TheWeave/prime_powers.dm:11]:
```suggestion
	if(!ishuman(human) || !human.allow_gun_usage)
```

File: code/modules/mob/living/carbon/xenomorph/abilities/TheWeave/prime_powers.dm [hunk 19]
```diff
@@ -0,0 +1,99 @@
...
+	human.update_canmove()
+	human.update_xeno_hostile_hud()
+
+	apply_cooldown()
+
+	self.frozen = 1
+	self.anchored = TRUE
+	self.update_canmove()
+
+	if (do_after(self, activation_delay, INTERRUPT_ALL | BEHAVIOR_IMMOBILE, BUSY_ICON_HOSTILE))
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/abilities/TheWeave/prime_powers.dm:42]: early return

File: code/modules/mob/living/carbon/xenomorph/abilities/TheWeave/prime_powers.dm [hunk 20]
```diff
@@ -0,0 +1,99 @@
...
+		if(nexus.use_energy(plasma_cost))
+			self.visible_message(SPAN_XENOHIGHDANGER("[self] floods [human]'s mind with The Weave!"), SPAN_XENOHIGHDANGER("You flood the mind of [human] with The Weave!"))
+
+			human.apply_effect(get_xeno_stun_duration(human, 0.5), WEAKEN)
+
+			human.WeaveClaim(CAUSE_WEAVER)
+		else
+			to_chat(self, SPAN_XENOWARNING("The Weave is not strong enough here to do that!"))
+
+	self.frozen = 0
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/abilities/TheWeave/prime_powers.dm:52]: if this is a boolean it should be using TRUE/FALSE

File: code/modules/mob/living/carbon/xenomorph/abilities/TheWeave/prime_powers.dm [hunk 21]
```diff
@@ -0,0 +1,99 @@
...
+			if(possible_target == self || !possible_target.client || !istype(possible_target))
+				continue
+			target_list += possible_target
+		for(var/mob/living/carbon/target in target_list)
+			if(HAS_TRAIT(target, TRAIT_WEAVE_SENSITIVE))
+				target.apply_damage(BRUTE, -40)
+				target.apply_damage(BURN, -40)
+				to_chat(target, SPAN_XENOBOLDNOTICE("You have been regenerated by the The Weave!"))
+			else
+				to_chat(target, SPAN_XENOBOLDNOTICE("You feel a ripple as energies pass over you!"))
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/abilities/TheWeave/prime_powers.dm:94]:
```suggestion
				to_chat(target, SPAN_XENOBOLDNOTICE("You have been regenerated by the The Weave!"))
				continue
			
			to_chat(target, SPAN_XENOBOLDNOTICE("You feel a ripple as energies pass over you!"))
```

File: code/modules/mob/living/carbon/xenomorph/abilities/TheWeave/prime_powers.dm [hunk 22]
```diff
@@ -0,0 +1,99 @@
...
+	else
+		self.visible_message(SPAN_XENOWARNING("[self] decides not to release their energy."), SPAN_XENOWARNING("You decide not to release your energy."))
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/abilities/TheWeave/prime_powers.dm:96]: early return this instead

File: code/modules/mob/living/carbon/xenomorph/abilities/TheWeave/weave_powers.dm [hunk 23]
```diff
@@ -0,0 +1,30 @@
...
+	if (do_after(self, 10 SECONDS, INTERRUPT_ALL | BEHAVIOR_IMMOBILE, BUSY_ICON_FRIENDLY))
+		if (!check_and_use_plasma_owner())
+			to_chat(self, SPAN_XENOWARNING("You do not have enough plasma stored to do this. You have [self.plasma_stored]/[plasma_cost]!"))
+			return FALSE
+
+		self.visible_message(SPAN_XENONOTICE("[self] exudes energy back into The Weave!"), SPAN_XENONOTICE("You release some of your energy into The Weave!"))
+
+		nexus.weave_energy += (plasma_cost / 10)
+	else
+		self.visible_message(SPAN_XENOWARNING("[self] decides not to release their energy."), SPAN_XENOWARNING("You decide not to release your energy."))
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/abilities/TheWeave/weave_powers.dm:27]: early return this instead

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm [hunk 24]
```diff
@@ -1132,6 +1133,32 @@
...
+		return FALSE
+	else
+		return TRUE
+
+/datum/hive_status/mutated/weave/proc/use_energy(cost)
+	if((weave_energy - cost) < 0)
+		return FALSE
+	else
+		weave_energy -= cost
+		return TRUE
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:1160]:
```suggestion

	weave_energy -= cost
	return TRUE
```

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm [hunk 25]
```diff
@@ -1132,6 +1133,32 @@
...
+	hive_inherant_traits = list(TRAIT_XENONID, TRAIT_NO_COLOR, TRAIT_NO_PREFIX, TRAIT_WEAVE_SENSITIVE)
+
+	var/bioscan_time = 0
+	var/weave_energy = 1000
+
+/datum/hive_status/mutated/weave/proc/can_use_energy(cost)
+	if((weave_energy - cost) < 0)
+		return FALSE
+	else
+		return TRUE
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:1153]:
```suggestion

	return TRUE
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by Watermelon914 — CHANGES_REQUESTED** [review]
Needs a proper description, justification and changelog.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Pull request closed without merging**